### DISCUSSION TITLE
Celery worker is emitting the message to a specified room but nothing received on UI.

### DISCUSSION THREAD

**Discussion by Mr-Dj-Python** [miguelgrinberg/Flask-SocketIO#1823] [Q&A] [answered]
Hello @miguelgrinberg 
I stuck with a new issue where
 new updates are received by any 3rd party application that is processed by the Celery worker which is reading the message from RabbitMQ 
once workers process information successfully all the process is done it should broadcast a message to all the client which are connected to the specified room.
please advise me something on this
## extension.py
added message_queue and other necessary information
![Screenshot from 2022-05-01 00-51-09](https://user-images.githubusercontent.com/72373042/166120249-5fa89104-cb7a-448c-b24c-3b8cfaae1ee0.png)

## app.py
![Screenshot from 2022-05-01 00-51-26](https://user-images.githubusercontent.com/72373042/166120270-1b050408-0edb-4460-846d-d012040ec359.png)

## socket methods.py
this is the main file for UI to communicate and join the room for sending event any update happened in terms of UI  changes
 
![Screenshot from 2022-05-01 00-51-52](https://user-images.githubusercontent.com/72373042/166120348-c55ca1be-e7fa-43fb-b61a-7a649b025b9e.png)

## worker broadcast function 

**it is sending information without any error but the message is not received by UI** 
![Screenshot from 2022-05-01 00-50-56](https://user-images.githubusercontent.com/72373042/166120400-2658f0b0-2af1-4d4f-9bb6-639367dac556.png)

![Screenshot from 2022-05-01 01-19-02](https://user-images.githubusercontent.com/72373042/166120569-27d14c58-cd52-4b18-962c-ca3397e0f6d0.png)

**Comment by miguelgrinberg** [comment, marked as answer]
See the troubleshooting section of the documentation to learn how to enable logs. The logs will show you all the incoming and outgoing network activity. This will help you determine if the event is being sent or not.

**Reply by Mr-Dj-Python** [reply]
I followed the document and have added monkey mentioned below image
![image](https://user-images.githubusercontent.com/72373042/169004849-5aa3596a-0dcf-4813-b117-c89ffcd94c2d.png)

also if I remove the message_queue attribute so everything works fine.
do I need to add something in the celery app context?

**Reply by miguelgrinberg** [reply]
You are ignoring what I asked. Please enable logs to see what events are being sent and/or received.

**Reply by Mr-Dj-Python** [reply]
I did that too forgot to mention it in the update 

I am receiving logs emitting messages from UI

![screenshot-nimbusweb me-2022 05 18-15_12_56](https://user-images.githubusercontent.com/72373042/169010346-1d0bb6e0-42d0-43e0-938e-6a58cbef2a57.png)

it receive the message but not able to emit

**Reply by miguelgrinberg** [reply]
If you don't see the things that you are emitting, then you are not calling the emit function. The log is written when you call the function, not when the other side receives your data. I suggest you debug your server-side application to make sure you call emit.

**Reply by Mr-Dj-Python** [reply]
<img width="597" alt="Screenshot 2022-05-19 at 1 20 47 PM" src="https://user-images.githubusercontent.com/72373042/169241644-5f6b8472-1bf9-4018-9cfa-5fff1ab35653.png">
 from above screen join event send a event so it should send event 
 although I move message_queue attribute from SocketIO object to init_app methods now eveything works fine.

**Reply by Mr-Dj-Python** [reply]
Thanks for the guidence
